tajo-catalog/tajo-catalog-common/src/main/java/org/apache/tajo/catalog/TableDesc.java

```diff
@@ -35,34 +36,43 @@
 
 public class TableDesc implements ProtoObject<TableDescProto>, GsonObject, Cloneable {
 	@Expose protected String tableName;                        // required
-  @Expose protected Schema schema;
+  @Expose protected Schema schema;                           // optional for self-describing tables
   @Expose protected TableMeta meta;                          // required
   /** uri is set if external flag is TRUE. */
-  @Expose protected URI uri;                                // optional
+  @Expose protected URI uri;                                 // optional
```

Comment:
In the last work, I changed URI to be required. Regardless of whether it is manged or external, every TableDescs has the table URI. The evidence is unused ``TableDesc::hasUri()``, which can be removed. Additional evidence is that there is no nullity check around ``getUri()``.

Otherwise, it is a bug.